spark/Spark-thirtserver-kyroserialization-exception.md

```diff
@@ -0,0 +1,48 @@
+---
+title: Unable to download/ retrieve large files when interactive the SparkSQL using Thrift Server? | Microsoft Docs
+description: Use the Spark FAQ for answers to common questions on Spark on Azure HDInsight platform.
+keywords: Azure HDInsight, Thrift Server, FAQ, troubleshooting guide, common problems, default serialization, kyro, remote submission
+services: Azure HDInsight
+documentationcenter: na
+author: Sunilkc
+manager: ''
+editor: ''
+
+ms.assetid: 
+ms.service: multiple
+ms.workload: na
+ms.tgt_pltfrm: na
+ms.devlang: na
+ms.topic: article
+ms.date: 07/28/2017
+ms.author: sunilkc
+---
+
+
+### Why do I see Spark Exception stating Kyro Serialization failed, similar to the one below, when trying to download/ retrieve a large data via JDBC/ODBC using Thrift Server. This issue is most likely observed on clusters using spark version  2.0.0 or earlier.
+
+~~~~
+ org.apache.spark.SparkException: Kryo serialization failed: 
+ Buffer overflow. Available: 0, required: 36518. To avoid this, increase spark.kryoserializer.buffer.max value.
+~~~~
+
+#### Kyro is a Object Graph Serialization framework for Java, similar to Apache Avro or Google's Protobuf.
+#### We usually start making changes to the spark.kyroserializer.buffer.max, somewhere in one of the spark versions its clearly documented the maximum value that you can set  is 2048m.
```

Comment:
This line is not clear as to what the customer suppose to do. In general we will be just prescriptive.